src/Module/GetBookingDisplayTimezoneCapableTrait.php

```diff
@@ -106,7 +106,11 @@ protected function _getDisplayTimezone($bookingData)
      */
     protected function _getBookingClientTimezone($bookingData)
     {
-        if (!$this->_containerHas($bookingData, 'client_tz')) {
+        $clientTz = $this->_containerHas($bookingData, 'client_tz')
+            ? $this->_containerGet($bookingData, 'client_tz')
+            : null;
+
+        if (empty($clientTz)) {
```

Comment:
I think that this is an opportunity for a bug, similar to what lead to this PR in the first place. It's the `null` value that indicates the absence of a timezone. `""` is a present timezone, but it is invalid, and should be treated that way.

Reply:
There is nothing that says that `""` is a present timezone. In fact, for most data models with known schemas, non-present values have meaningful defaults. This is actually the case with EDD Bookings, where a booking retrieved from the database will have the `client_tz` key present, but will have a `""` value. Or maybe a `NULL` value. I don't recall how I set up that table. But it shouldn't matter.

Reply:
`""` is a value. `null` is a value that specifically indicates the absence of a value. The code should attempt to create a timezone object with anything that is a present value. And it should fail if that value is invalid.

However, if the underlying data happens to be using `""` as a default to mean "no value" instead of `NULL`, then I will accept. But please confirm.

Reply:
Confirmed.

Reply:
What about other empty values? `0` should throw, not return `null`.

Reply:
Same for `"0"`.

Reply:
> should throw
Rather, whether to throw should be up to the constructor of the date-time object. That's what I meant.

Reply:
Ah, good point. That should definitely be classified as an invalid timezone.

What do you think about utilizing PHP's `null` to `string` casting for an empty string check?

```
if (strlen((string)$tz) === 0) return null;
```

Reply:
_Aside_: I'm trying to find the best way to catch both empty strings and nulls, without having to go old school.

Reply:
That sounds much better.

Reply:
I think I hit my head or something.

Reply:
Happy days! [`strlen` accepts `null` and returns `0`.](https://3v4l.org/0s1Gs)